File: src/main/java/de/difuture/ekut/pht/lib/core/neo4j/converter/URIConverter.java
package de.difuture.ekut.pht.lib.core.neo4j.converter;

import org.neo4j.ogm.typeconversion.AttributeConverter;

import java.net.URI;


/**
 *
 * @author Lukas Zimmermann
 */
public final class URIConverter implements AttributeConverter<URI, String> {

    private static final String NA = "NA";

    @Override
    public String toGraphProperty(URI value) {

        if (value == null) {

            return NA;
        }
        return value.toASCIIString();
    }

    @Override
    public URI toEntityAttribute(String value) {

        if (value == null || NA.equals(value)) {

            return null;
        }
        return URI.create(value);
    }
}

File: src/main/kotlin/de/difuture/ekut/pht/entity/converter/URIConverter.kt
package de.difuture.ekut.pht.entity.converter

import de.difuture.ekut.pht.lib.core.constants.NA
import org.neo4j.ogm.typeconversion.AttributeConverter
import java.net.URI


class URIConverter : AttributeConverter<URI, String> {

    override fun toGraphProperty(value: URI?): String {

        return value?.toASCIIString() ?: NA
    }


    override fun toEntityAttribute(value: String?): URI? {

        return if (value == null || NA == value) {

            null

        } else URI.create(value)
    }
}
